File: flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/theoplayer/transformers/PlayerEnumTransformer.kt
package com.theoplayer.theoplayer.transformers

import com.theoplayer.android.api.player.PreloadType
import com.theoplayer.android.api.player.ReadyState

typealias FlutterReadyState = com.theoplayer.theoplayer.pigeon.ReadyState
typealias FlutterPreloadType = com.theoplayer.theoplayer.pigeon.PreloadType

object PlayerEnumTransformer {

    fun toFlutterReadyState(readyState: ReadyState): FlutterReadyState {
        return when (readyState) {
            ReadyState.HAVE_NOTHING -> FlutterReadyState.HAVE_NOTHING
            ReadyState.HAVE_METADATA -> FlutterReadyState.HAVE_METADATA
            ReadyState.HAVE_CURRENT_DATA -> FlutterReadyState.HAVE_CURRENT_DATA
            ReadyState.HAVE_FUTURE_DATA -> FlutterReadyState.HAVE_FUTURE_DATA
            ReadyState.HAVE_ENOUGH_DATA -> FlutterReadyState.HAVE_ENOUGH_DATA
        }
    }

    fun toFlutterPreloadType(preload: PreloadType): FlutterPreloadType {
        return when (preload) {
            PreloadType.NONE -> FlutterPreloadType.NONE
            PreloadType.AUTO -> FlutterPreloadType.AUTO
            PreloadType.METADATA -> FlutterPreloadType.METADATA
        }
    }

    fun toPreloadType(preload: FlutterPreloadType): PreloadType {
        return when (preload) {
            FlutterPreloadType.NONE -> PreloadType.NONE
            FlutterPreloadType.AUTO -> PreloadType.AUTO
            FlutterPreloadType.METADATA -> PreloadType.METADATA
        }
    }
}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/Classes/transformers/PlayerEnumTransformer.swift
//
//  PlayerEnumTransformer.swift
//  flutter_theoplayer_sdk_ios
//
//  Created by Hovig on 16/10/2023.
//

import Foundation
import THEOplayerSDK

struct PlayerEnumTransformer {
    
    static func toFlutterReadyState(readyState: THEOplayerSDK.ReadyState) -> ReadyState {
        switch(readyState) {
        case .HAVE_NOTHING:
            return .haveNothing
        case .HAVE_METADATA:
            return .haveMetadata
        case .HAVE_CURRENT_DATA:
            return .haveCurrentData
        case .HAVE_FUTURE_DATA:
            return .haveFutureData
        case .HAVE_ENOUGH_DATA:
            return .haveEnoughData
        default:
            return .haveNothing
        }
    }
    
    static func toFlutterPreloadType(preload: THEOplayerSDK.Preload) -> PreloadType {
        switch(preload) {
        case .none:
            return .none
        case .auto:
            return .auto
        case .metadata:
            return .metadata
        default:
            return .none
        }
    }
    
    static func toPreloadType(preload: PreloadType) -> THEOplayerSDK.Preload {
        switch(preload) {
        case .none:
            return .none
        case .auto:
            return .auto
        case .metadata:
            return .metadata
        }
    }
    
}
